ui/selectors/assets.ts

```diff
@@ -40,3 +59,152 @@ export function getAssetsMetadata(state: AssetsState) {
 export function getAssetsRates(state: AssetsRatesState) {
   return state.metamask.conversionRates;
 }
+
+export const getTokenBalancesEvm = createDeepEqualSelector(
```

Comment:
Can this selector return an icon for each token as well?

Reply:
I think this one should already return the `image` of the token; were you maybe referencing sth else?

Reply:
Doesn't seem like it, this is one example:
```
{"symbol":"ETH","decimals":18,"name":"Ethereum Mainnet","address":"","balance":"0","chainId":"0x1","isNative":true,"tokenFiatAmount":0,"string":"0","primary":"","secondary":0,"title":""}
```

Reply:
Oh right, the images we get for ERC20 come from token-api, but for the native token we do not return the image, Today we use `getNativeCurrencyForChain` to display the native token images; probably we can add that to the selector

Reply:
Pushed an update here [cd357c0](https://github.com/MetaMask/metamask-extension/pull/30389/commits/cd357c07183dd3bec6d38caa4e3cf1f35f594dc1) 🙏